### PR TITLE
Add SerialPortToLog parameter to InstanceBase

### PR DESCRIPTION
**Pull request by same-id**
When using the WaitForInstancesSignal with a port other than 1:

  "WaitForInstancesSignal": [{
    "Name": "builder",
    "SerialOutput": {
      "Port": 3,
      "SuccessMatch": "BuilderSuccess:",
      "FailureMatch": "BuilderFailure:",
      "StatusMatch": "BuilderStatus:"
    }
  }]

It is not possible to see the entire serial port logs in GCS due to the
fact that the logging is actually an attribute of the CreateInstances
and the port number there is hardcoded to 1.

Introduce SerialPortToLog parameter for instance base to configure which
serial port is being logged for the given instance.

### CODE REVIEW

**Review by zoran15 — COMMENTED** [review]
Add test coverage to instance_test.go and step_create_instance_test.go

**Comment by same-id** [comment]
Added test cases

File: daisy/instance.go [hunk 1]
```diff
@@ -632,6 +643,13 @@ func (i *InstanceBeta) getComputeDisks() []*computeDisk {
 	return computeDisks
 }
 
+func (ib *InstanceBase) validateSerialPortToLog() (errs DError) {
+	if ib.SerialPortToLog < 0 || ib.SerialPortToLog > 4 {
+		errs = addErrs(errs, Errf("cannot create instance: SerialPortToLog must be between 1-4"))
```
Comment (EricEdens) [inline daisy/instance.go:648]:
```suggestion
		errs = addErrs(errs, Errf("cannot create instance: SerialPortToLog must be between 1-4, inclusive"))
```

**Comment by zoran15** [comment]
> Added test cases

I don't see the change. Maybe you didn't push?

**Review by EricEdens — COMMENTED** [review]
thanks for the PR! lgtm (ditto Zoran's feedback about tests)

**Comment by same-id** [comment]
> > Added test cases
> 
> I don't see the change. Maybe you didn't push?

Yep, my bad

File: daisy/instance.go (hunk 1 above)
Comment (same-id) [reply daisy/instance.go]: Added

**Review by EricEdens — APPROVED** [review]

**Review by zoran15 — APPROVED** [review]

File: daisy/instance.go [hunk 2]
```diff
@@ -106,10 +106,12 @@ type InstanceBase struct {
 	// Should an existing instance of the same name be deleted, defaults to false
 	// which will fail validation.
 	OverWrite bool `json:",omitempty"`
+	// Serial port to log to GCS bucket, defaults to 1
+	SerialPortToLog int64 `json:",omitempty"`
```
[the remark this answers is not in the document]
Comment (same-id) [reply daisy/instance.go]: Sure, done

**Review by adjackura — APPROVED** [review]

**Pull request merged**